src/env.sh

```diff
@@ -57,33 +51,25 @@ env_init_pkg() {
     # Check if the package exists/has an ellipsis.sh file
     if [ -f "$PKG_PATH/ellipsis.sh" ]; then
         # Check if the package has an init hook
-        grep '^pkg.init' "$PKG_PATH/ellipsis.sh" 2>&1 >/dev/null
-        if [ $? -eq 0 ]; then
-            # Use some magic to extract the init function
-            pkg_init="$(bash -c "source $PKG_PATH/ellipsis.sh; declare -f pkg.init")"
-            pkg_init="$(echo "$pkg_init" | sed 's/pkg.init/pkg_init/')"
+        pkg_init="$(sed -n '/^pkg.init\(\)/,/^}/p' "$PKG_PATH/ellipsis.sh" | sed '1 s/\./_/')"
```

Comment:
This will need a few more comments so everyone knows what's going on.

Have to do a few more tests on this to see if it handles all cases. The env functions could use some more automatic tests. Would be nice if it works reliably, as the bash trick was quite ugly and relatively slow.

Reply:
The only limitation of the first sed is that `pkg.init` and the `}` should be at the beginning of the line. It prints everything from `^pkg.init` to the first matching `^}`. The second sed replaces `.` by `_` on the first line only.

Reply:
2ms / package on my computer (slower than yours apparently 😄 )